From 4276f9752068aa4adaa7dcd037b3d7805bb51bc0 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:37 -0400
Subject: [infiniband] mthca: Use pci_enable_msix_exact() instead of pci_enable_msix()

Message-id: <07ae8b70428258d7b98bbc2fdc21806c3ec8c0b0.1410425016.git.avadai@redhat.com>
Patchwork-id: 93679
O-Subject: [Patch RHEL7.1 V1 101/197] IB/mthca: Use pci_enable_msix_exact() instead of pci_enable_msix()
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of the deprecation of the MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block(), all drivers using these
two interfaces need to be updated to use the new
pci_enable_msi_range() or pci_enable_msi_exact() and
pci_enable_msix_range() or pci_enable_msix_exact() interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 9684c2ea6d1f5aab44119533530e4059b4c3e1ff)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mthca/mthca_main.c b/drivers/infiniband/hw/mthca/mthca_main.c
index 87897b9..ded76c1 100644
--- a/drivers/infiniband/hw/mthca/mthca_main.c
+++ b/drivers/infiniband/hw/mthca/mthca_main.c
@@ -858,13 +858,9 @@ static int mthca_enable_msi_x(struct mthca_dev *mdev)
  entries[1].entry = 1;
  entries[2].entry = 2;
 
- err = pci_enable_msix(mdev->pdev, entries, ARRAY_SIZE(entries));
- if (err) {
-  if (err > 0)
-   mthca_info(mdev, "Only %d MSI-X vectors available, "
-       "not using MSI-X\n", err);
+ err = pci_enable_msix_exact(mdev->pdev, entries, ARRAY_SIZE(entries));
+ if (err)
   return err;
- }
 
  mdev->eq_table.eq[MTHCA_EQ_COMP ].msi_x_vector = entries[0].vector;
  mdev->eq_table.eq[MTHCA_EQ_ASYNC].msi_x_vector = entries[1].vector;
-- 
1.7.1